### PR TITLE
Remove fixed GA date map in favor of cached lookup

### PR DESCRIPTION
**Pull request by dgoodwin**
Found this when I flipped the view to use 4.17 ga which wasn't added to the map as we didn't have a step before.

We believe this is safe to assume bigquery connectivity now. I think the fallback to postgres was probably broken.

Added caching for speed.

### CODE REVIEW

File: pkg/sippyserver/metrics/installupgradesuccess.go [hunk 1]
```diff
@@ -3,14 +3,14 @@ package metrics
 import (
 	"math"
 
+	v12 "github.com/openshift/sippy/pkg/apis/sippy/v1"
```
Comment (dgoodwin) [inline pkg/sippyserver/metrics/installupgradesuccess.go:6]: auto refactor, fixing

**Review by sosiouxme — APPROVED** [review]
/lgtm
but
/hold
because i'm not sure if you want to YOLO on a friday afternoon

**Comment by dgoodwin** [comment]
lint failing on npm security fixes, we will address separately.

/override ci/prow/lint

**Pull request merged**